### PR TITLE
🪑 Create table to manage resources

### LINKED ISSUE #382: Incorporate sidebar into resource table
## Description
Link sidebar and resource table so that we can filter the table in the resource manager

### PR DESCRIPTION
**Pull request by Feeshay**
## Status:
:rocket: Ready

## Description

* Creates a table with all resources
* Displays (sub)categories as tags for each resource
* Link to edit each resource
* Integrate with category sidebar (mostly works)

Part of #357 
Resolves #382 

## Todos
- [x] Convert categories and stuff to tags
- [x] Route to edit resource page
- [ ] Add functionality to remove (sub)categories (another PR)
- [ ] Add functionality to add (sub)categories (another PR)
- [ ] Make endpoints more efficient and only query the info we need (yet another PR)

## Screenshots
image

Filtering:
image

### CODE REVIEW

File: client/src/components/ManageResourcesTable.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  x: 'purple',
+  y: 'purple',
+  z: 'purple',
+};
+
+const ManageResourcesTable = () => {
+  const [resources, setResources] = useState([]);
+
+  const displayTags = (categories) => (
+    <div>
```
Comment (jackieo5023) [inline client/src/components/ManageResourcesTable.jsx:40]: We can use a fragment for this (unless that breaks something)

File: client/src/components/ManageResourcesTable.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        style={{ cursor: 'pointer' }}
+      >
+        +
+      </Tag>
+    </div>
+  );
+
+  useEffect(() => {
+    const fetchResources = async () => {
+      // TODO: replace with getResourcesByCategory and getResourcesBySubcategory
```
Comment (jackieo5023) [inline client/src/components/ManageResourcesTable.jsx:57]: wanna make a github issue for these :))

File: client/src/components/RoleApproval.jsx [hunk 3]
```diff
@@ -145,10 +145,7 @@ const RoleApproval = () => {
           }}
         />
       ) : (
-        <Card
-          className="interview-card"
-          style={{ height: '60%', margin: '2%' }}
-        >
+        <>
```
Comment (jackieo5023) [inline client/src/components/RoleApproval.jsx:148]: Wait what was this lol (also if it's unneeded let's also fix that linting comment)

**Review by jackieo5023 — APPROVED** [review]
LGTM - make sure to address the 3 comments, get another review, and shipit!

File: client/src/components/RoleApproval.jsx (hunk 3 above)
Comment (Feeshay) [reply client/src/components/RoleApproval.jsx:148]: The card was adding a border around the role approval when it was its own page but now there's an admin portal so...

File: client/src/components/ManageResourcesTable.jsx (hunk 2 above)
Comment (Feeshay) [reply client/src/components/ManageResourcesTable.jsx:57]: ugh fine

File: client/src/components/ManageResourcesTable.jsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
+// @flow
+
+import React, { useState, useEffect } from 'react';
+import { EditFilled } from '@ant-design/icons';
+import { Table, Tag } from 'antd';
+
+import { getResources } from '../utils/api';
+
+const TAG_COLOR_DICT = {
```
Comment (n3a9) [inline client/src/components/ManageResourcesTable.jsx:9]: Might be better to put this in `src/utils` file and then import the dictionary just to clean the file. But definitely a design choice.

**Review by n3a9 — APPROVED** [review]
beautiful

**Pull request merged**